### PR TITLE
[ISM][#18651] TierPrice save float percentage value

### PR DESCRIPTION
**Pull request by ghost**
### Description
Problem:
After save tierprice for product, `discount` field lost value after comma.
Fix:
Updated getPercentage method in TierPrices:
- Type conversation changed: int -> float
- Added strict types
- Database has remained unchanged since the percentage_value is a decimal field

### Fixed Issues
1. magento/magento2#18651: Tierprice can't save float percentage value

### Manual testing scenarios
1. Add new product
2. Add new tierprice with `discount` value "2.5"
3. Save the product
4. Value "2.5" is saved

### Contribution checklist
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds on Travis CI are green)

### CODE REVIEW

**Comment by magento-cicd2** [comment]
[CLA assistant check](https://cla.dev.magento.com/magento/magento2?pullRequest=19030) <br/>All committers have signed the CLA.

**Comment by magento-engcom-team** [comment]
Hi @SikailoISM. Thank you for your contribution 
Here is some useful tips how you can test your changes using Magento test environment.
Add the comment under your pull request to deploy test or vanilla Magento instance:
- `@magento-engcom-team give me test instance` - deploy test instance based on PR changes
- `@magento-engcom-team give me $VERSION instance` - deploy vanilla Magento instance

For more details, please, review the [Magento Contributor Assistant](https://devdocs.magento.com/guides/v2.2/contributor-guide/contributing.html#contributor-assist) documentation

File: app/code/Magento/Catalog/Model/Product/Attribute/Backend/TierPrice/SaveHandler.php
Comment (vadimjustus) [inline app/code/Magento/Catalog/Model/Product/Attribute/Backend/TierPrice/SaveHandler.php]: Hi @SikailoISM, since Magento 2.2 supports PHP 7.0 you can't use nullable return types as this is a feature of PHP 7.1 and higher. See http://php.net/manual/en/migration71.new-features.php

File: app/code/Magento/Catalog/Model/Product/Attribute/Backend/TierPrice/UpdateHandler.php
Comment (vadimjustus) [inline app/code/Magento/Catalog/Model/Product/Attribute/Backend/TierPrice/UpdateHandler.php]: Hi @SikailoISM, since Magento 2.2 supports PHP 7.0 you can't use nullable return types as this is a feature of PHP 7.1 and higher. See http://php.net/manual/en/migration71.new-features.php

**Review by vadimjustus — CHANGES_REQUESTED** [review]

**Comment by tkacheva** [comment]
@SikailoISM Thanks for contributing this change! That's great to have tier discounts format aligned with catalog and cart promotions discounts

File: app/code/Magento/Catalog/Model/Product/Attribute/Backend/TierPrice/SaveHandler.php
Comment (orlangur) [inline app/code/Magento/Catalog/Model/Product/Attribute/Backend/TierPrice/SaveHandler.php]: There is no need to cast explicitly to `float`, usually `+ 0` or `* 1` is used to convert so that no unneeded int <-> float occur.

**Review by orlangur — CHANGES_REQUESTED** [review]
Nowadays all changes must be applied to `2.3-develop` first. Please prepare a new pull request, this one should be `on hold` until the latter is merged.

File: app/code/Magento/Catalog/Model/Product/Attribute/Backend/TierPrice/SaveHandler.php
Comment (ghost) [reply app/code/Magento/Catalog/Model/Product/Attribute/Backend/TierPrice/SaveHandler.php]: • $priceRow['percentage_value'] + 0 or • $priceRow['percentage_value'] * 1
Is it relevant?

Comment (orlangur) [reply app/code/Magento/Catalog/Model/Product/Attribute/Backend/TierPrice/SaveHandler.php]: Yes, of course.

**Comment by sidolov** [comment]
Hi @SikailoISM , please, resolve merge conflicts

**Comment by ghost** [comment]
I give up

**Pull request closed without merging**
